data_structures/linked_list/linked_list.py

```diff
@@ -184,6 +184,24 @@ def kthFromEnd(self, k):
             node = node._next
         raise IndexError('requested node outside linked list length')
 
+    def hasLoop(self):
+        """returns false if no looped node reference"""
+        self.node1 = self.head
+        if self.node1 is None:
+            return False
+        else:
```

Comment:
Since the `if` statement returns a value, you do not need to actually write an `else` statement.